routes/settings.go

```diff
@@ -364,22 +365,37 @@ func (h *SettingsHandler) actionAddLabel(w http.ResponseWriter, r *http.Request)
 	}
 	user := middlewares.GetPrincipal(r)
 
-	label := &models.ProjectLabel{
-		UserID:     user.ID,
-		ProjectKey: r.PostFormValue("key"),
-		Label:      r.PostFormValue("value"),
-	}
+	var labels []*models.ProjectLabel
 
-	if !label.IsValid() {
-		return http.StatusBadRequest, "", "invalid input"
+	if r.PostFormValue("num_projects") == "multiple" {
```

Comment:
I think you don't even need this distinction here. You can probably just do `<select name="key">` and both will get parsed as a list, while in the first case (non-batch associate), it will only have one entry.